shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1349,14 +1370,19 @@ - (void)startKeyBoardAnimation:(NSTimeInterval)duration {
 
   if ([self keyboardAnimationView].superview == nil) {
     [self.view addSubview:[self keyboardAnimationView]];
+  } else if (self.keyboardAnimationIsCompounding) {
+    // If keyboard animation is in the same direction as current animation, ignore it.
+    return;
   }
```

Comment:
Should be 

```objc
if ([self keyboardAnimationView].superview == nil) {
    [self.view addSubview:[self keyboardAnimationView]];
} 

if (self.keyboardAnimationIsCompounding) {
    // If keyboard animation is in the same direction as current animation, ignore it.
    return;
}
```
Remove `else`, I think they shouldn't be mutually exclusive

Reply:
Yeah I was thinking about that, but the else would ensure that it only escapes when there is a keyboardAnimationView (an animation in progress)